include/CLI/Option.hpp

```diff
@@ -745,6 +791,74 @@ class Option : public OptionBase<Option> {
     /// Get a copy of the results
     std::vector<std::string> results() const { return results_; }
 
+    /// get the results as a particular type
+    template <typename T,
+              enable_if_t<!is_vector<T>::value && !std::is_const<T>::value, detail::enabler> = detail::dummy>
+    void results(T &output) const {
```

Comment:
Interesting! I've been looking for a way to add types to Option for 2.0, but this partially (not fully) negates the need to do that. The tricky part about the 2.0 idea is that `Option<T>` would need to be stored in the shared/unique pointer array, and I was thinking it would need a reference or copy of T, but this might avoid that (at a cost per access).

Does this pick up the results of a transform? I assume so since they should be transformed in place?

Reply:
Could this be broken out into a standalone function(s including overload)? Then it could be reused by the normal parse also, possibly? That is, this would become something like:
```
template<typename T>
void results(T &output) {
    CLI::detail::results(output, results_, defaultval_, multi_option_policy_);
}
```
And `CLI::detail::results` could be used in the parse function, too. Maybe not, just thinking. Also, it might be possible to make `CLI::detail::results` a struct, and it could have things like default number of arguments attached, and a user could provide specializations to add support for new types. (now well outside the scope of this PR, just thinking)